presto-spark-base/src/main/java/com/facebook/presto/spark/execution/AbstractPrestoSparkQueryExecution.java

```diff
@@ -573,6 +632,7 @@ protected void queryCompletedEvent(Optional<ExecutionFailureInfo> failureInfo, O
             byte[] bytes = serializedTaskInfo.getBytesAndClear();
             totalSerializedTaskInfoSizeInBytes += bytes.length;
             TaskInfo taskInfo = deserializeZstdCompressed(taskInfoCodec, bytes);
+            collectMetrics(taskInfo);
```

Comment:
@miaoever MJ, why do we need to collect both taskInfo and metrics derived from it? Should we derive metrics on the driver instead?

Reply:
> @miaoever MJ, why do we need to collect both taskInfo and metrics derived from it? Should we derive metrics on the driver instead?

We need to derive here because in Presto-on-spark library, on driver side we can only  aggregate stage level metrics. For task level metrics, we need to derive the metrics on executor here and aggregate them during task update event inside spark.